4주차/18232/java_18232_panda.java

```diff
@@ -0,0 +1,69 @@
+package algorithm_study.week4;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+import java.util.Set;
+import java.util.StringTokenizer;
+import java.util.stream.Stream;
+
+public class java_18232_panda {
+    static int N, M, S, E;
+    static Map<Integer, Set<Integer>> teleport;
+    static boolean[] visited;
+
+    public static void main(String[] args) throws IOException {
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        StringTokenizer st = new StringTokenizer(br.readLine());
+        N = Integer.parseInt(st.nextToken());
+        M = Integer.parseInt(st.nextToken());
+
+        st = new StringTokenizer(br.readLine());
+        S = Integer.parseInt(st.nextToken());
+        E = Integer.parseInt(st.nextToken());
+
+        visited = new boolean[N + 1];
+        teleport = new HashMap<>();
+        for (int i = 0; i < M; i++) {
+            st = new StringTokenizer(br.readLine());
+            int firstSpot = Integer.parseInt(st.nextToken());
+            int secondSpot = Integer.parseInt(st.nextToken());
+            teleport.computeIfAbsent(firstSpot, k -> new HashSet<>()).add(secondSpot);
```

Comment:
혹시 k를 선언하는 이유가 무엇일까요?
저는 아래처럼 사용하고 싶었는데 노란색 밑줄로 경고가 뜨고 k로 바꾸니까 다시 경고가 사라지더라고요💦

```suggestion
            teleport.computeIfAbsent(firstSpot, () -> new HashSet<>()).add(secondSpot);
```

Reply:
이게 사실 저도 궁금해서 찾아본 적이 있는데, `함수형 인터페이스의 타입을 추론할 수 없다` 라는 오류가 발생합니다.
그러니까 computeIfAbsent의 두 번째 인자로 `Function` 타입의 함수형 인터페이스를 받게 되어있는데, 아래처럼 argument 없이 람다식을 넘기면 컴파일러가 이게 어떤 타입인지 해석하지 못한다는 뜻인 것 같네요.